### PR TITLE
DHFPROD-5358: Fixing the issue where UI throws error when source document is not an entity instance

### PR DESCRIPTION
**Pull request by xnikhil08**
### Description

This PR fixes the bug DHFPROD-5358. Just a small bug fix so no unit tests added. @timur-isangulov will add an e2e test for this later.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by srinathgit** [comment]
With the latest code, I get this when I explore "Customer" entity

```
react-dom.development.js:13413 Uncaught Error: Objects are not valid as a React child (found: object with keys {undefined}). If you meant to render a collection of children, use an array instead.
    in div (at results-tabular-view.tsx:140)
    in Trigger (created by Tooltip)
    in Tooltip (created by Context.Consumer)
    in Tooltip (created by MLTooltip)
    in MLTooltip (at results-tabular-view.tsx:138)
    in td (created by TableCell)
    in TableCell (created by TableRow)
    in tr (created by BodyRow)
    in BodyRow (created by TableRow)
    in TableRow (created by Connect(TableRow))
    in Connect(TableRow) (created by ExpandableRow)
    in ExpandableRow (created by Connect(ExpandableRow))
    in Connect(ExpandableRow) (created by BaseTable)
    in tbody (created by BaseTable)
    in table (created by BaseTable)
    in BaseTable (created by Connect(BaseTable))
    in Connect(BaseTable) (created by BodyTable)
    in div (created by BodyTable)
    in BodyTable (created by ExpandableTable)
    in div (created by ExpandableTable)
    in div (created by ExpandableTable)
    in ExpandableTable (created by Connect(ExpandableTable))
    in Connect(ExpandableTable) (created by Table)
    in Provider (created by Table)
    in Table (created by LocaleReceiver)
    in LocaleReceiver (created by Context.Consumer)
    in div (created by Context.Consumer)
    in div (created by Context.Consumer)
    in Spin (created by Context.Consumer)
    in div (created by Context.Consumer)
    in Table (created by withStore(Table))
    in withStore(Table) (created by MLTable)
    in MLTableContextProvider (created by MLTable)
    in MLTable (at results-tabular-view.tsx:327)
    in div (at results-tabular-view.tsx:326)
    in ResultsTabularView (at Browse.tsx:272)
    in div (at Browse.tsx:271)
    in div (at Browse.tsx:269)
    in main (created by Basic)
    in Basic (created by Context.Consumer)
    in Content (at Browse.tsx:231)
    in section (created by BasicLayout)
    in BasicLayout (created by Context.Consumer)
    in Layout (at Browse.tsx:221)
    in Browse (created by Context.Consumer)
    in withRouter(Browse) (at TilesView.tsx:34)
    in div (created by InternalMosaicWindow)
    in div (created by InternalMosaicWindow)
    in InternalMosaicWindow (created by Context.Consumer)
    in DragSource(InternalMosaicWindow) (created by Context.Consumer)
    in DropTarget(DragSource(InternalMosaicWindow)) (created by MosaicWindow)
    in MosaicWindow (at tiles.tsx:134)
    in div (created by MosaicRoot)
    in div (created by MosaicRoot)
    in MosaicRoot (created by MosaicWithoutDragDropContext)
    in div (created by MosaicWithoutDragDropContext)
    in MosaicWithoutDragDropContext (created by Mosaic)
    in Unknown (created by Mosaic)
    in Mosaic (at tiles.tsx:131)
    in Tiles (at TilesView.tsx:110)
    in div (at TilesView.tsx:108)
    in TilesView (at App.tsx:118)
    in Route (at App.tsx:34)
    in PrivateRoute (at App.tsx:117)
    in ModelingProvider (at App.tsx:98)
    in SearchProvider (at App.tsx:91)
    in Switch (at App.tsx:86)
    in div (at App.tsx:85)
    in main (at App.tsx:84)
    in div (at App.tsx:81)
    in App (created by Context.Consumer)
    in withRouter(App) (at src/index.tsx:12)
    in UserProvider (at src/index.tsx:11)
    in Router (created by BrowserRouter)
    in BrowserRouter (at src/index.tsx:10)
    at throwOnInvalidObjectType (react-dom.development.js:13413)
    at createChild (react-dom.development.js:13651)
    at reconcileChildrenArray (react-dom.development.js:13900)
    at reconcileChildFibers (react-dom.development.js:14305)
    at reconcileChildren (react-dom.development.js:16762)
    at updateHostComponent (react-dom.development.js:17302)
    at beginWork (react-dom.development.js:18627)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at beginWork$1 (react-dom.development.js:23203)
    at performUnitOfWork (react-dom.development.js:22157)
    at workLoopSync (react-dom.development.js:22130)
    at performSyncWorkOnRoot (react-dom.development.js:21756)
    at react-dom.development.js:11089
    at unstable_runWithPriority (scheduler.development.js:653)
    at runWithPriority$1 (react-dom.development.js:11039)
    at flushSyncCallbackQueueImpl (react-dom.development.js:11084)
    at flushSyncCallbackQueue (react-dom.development.js:11072)
    at discreteUpdates$1 (react-dom.development.js:21893)
    at discreteUpdates (react-dom.development.js:806)
    at dispatchDiscreteEvent (react-dom.development.js:4168)
```

**Comment by xnikhil08** [comment]
> With the latest code, I get this when I explore "Customer" entity
> 
> ```
> react-dom.development.js:13413 Uncaught Error: Objects are not valid as a React child (found: object with keys {undefined}). If you meant to render a collection of children, use an array instead.
>     in div (at results-tabular-view.tsx:140)
>     in Trigger (created by Tooltip)
>     in Tooltip (created by Context.Consumer)
>     in Tooltip (created by MLTooltip)
>     in MLTooltip (at results-tabular-view.tsx:138)
>     in td (created by TableCell)
>     in TableCell (created by TableRow)
>     in tr (created by BodyRow)
>     in BodyRow (created by TableRow)
>     in TableRow (created by Connect(TableRow))
>     in Connect(TableRow) (created by ExpandableRow)
>     in ExpandableRow (created by Connect(ExpandableRow))
>     in Connect(ExpandableRow) (created by BaseTable)
>     in tbody (created by BaseTable)
>     in table (created by BaseTable)
>     in BaseTable (created by Connect(BaseTable))
>     in Connect(BaseTable) (created by BodyTable)
>     in div (created by BodyTable)
>     in BodyTable (created by ExpandableTable)
>     in div (created by ExpandableTable)
>     in div (created by ExpandableTable)
>     in ExpandableTable (created by Connect(ExpandableTable))
>     in Connect(ExpandableTable) (created by Table)
>     in Provider (created by Table)
>     in Table (created by LocaleReceiver)
>     in LocaleReceiver (created by Context.Consumer)
>     in div (created by Context.Consumer)
>     in div (created by Context.Consumer)
>     in Spin (created by Context.Consumer)
>     in div (created by Context.Consumer)
>     in Table (created by withStore(Table))
>     in withStore(Table) (created by MLTable)
>     in MLTableContextProvider (created by MLTable)
>     in MLTable (at results-tabular-view.tsx:327)
>     in div (at results-tabular-view.tsx:326)
>     in ResultsTabularView (at Browse.tsx:272)
>     in div (at Browse.tsx:271)
>     in div (at Browse.tsx:269)
>     in main (created by Basic)
>     in Basic (created by Context.Consumer)
>     in Content (at Browse.tsx:231)
>     in section (created by BasicLayout)
>     in BasicLayout (created by Context.Consumer)
>     in Layout (at Browse.tsx:221)
>     in Browse (created by Context.Consumer)
>     in withRouter(Browse) (at TilesView.tsx:34)
>     in div (created by InternalMosaicWindow)
>     in div (created by InternalMosaicWindow)
>     in InternalMosaicWindow (created by Context.Consumer)
>     in DragSource(InternalMosaicWindow) (created by Context.Consumer)
>     in DropTarget(DragSource(InternalMosaicWindow)) (created by MosaicWindow)
>     in MosaicWindow (at tiles.tsx:134)
>     in div (created by MosaicRoot)
>     in div (created by MosaicRoot)
>     in MosaicRoot (created by MosaicWithoutDragDropContext)
>     in div (created by MosaicWithoutDragDropContext)
>     in MosaicWithoutDragDropContext (created by Mosaic)
>     in Unknown (created by Mosaic)
>     in Mosaic (at tiles.tsx:131)
>     in Tiles (at TilesView.tsx:110)
>     in div (at TilesView.tsx:108)
>     in TilesView (at App.tsx:118)
>     in Route (at App.tsx:34)
>     in PrivateRoute (at App.tsx:117)
>     in ModelingProvider (at App.tsx:98)
>     in SearchProvider (at App.tsx:91)
>     in Switch (at App.tsx:86)
>     in div (at App.tsx:85)
>     in main (at App.tsx:84)
>     in div (at App.tsx:81)
>     in App (created by Context.Consumer)
>     in withRouter(App) (at src/index.tsx:12)
>     in UserProvider (at src/index.tsx:11)
>     in Router (created by BrowserRouter)
>     in BrowserRouter (at src/index.tsx:10)
>     at throwOnInvalidObjectType (react-dom.development.js:13413)
>     at createChild (react-dom.development.js:13651)
>     at reconcileChildrenArray (react-dom.development.js:13900)
>     at reconcileChildFibers (react-dom.development.js:14305)
>     at reconcileChildren (react-dom.development.js:16762)
>     at updateHostComponent (react-dom.development.js:17302)
>     at beginWork (react-dom.development.js:18627)
>     at HTMLUnknownElement.callCallback (react-dom.development.js:188)
>     at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
>     at invokeGuardedCallback (react-dom.development.js:292)
>     at beginWork$1 (react-dom.development.js:23203)
>     at performUnitOfWork (react-dom.development.js:22157)
>     at workLoopSync (react-dom.development.js:22130)
>     at performSyncWorkOnRoot (react-dom.development.js:21756)
>     at react-dom.development.js:11089
>     at unstable_runWithPriority (scheduler.development.js:653)
>     at runWithPriority$1 (react-dom.development.js:11039)
>     at flushSyncCallbackQueueImpl (react-dom.development.js:11084)
>     at flushSyncCallbackQueue (react-dom.development.js:11072)
>     at discreteUpdates$1 (react-dom.development.js:21893)
>     at discreteUpdates (react-dom.development.js:806)
>     at dispatchDiscreteEvent (react-dom.development.js:4168)
> ```

@srinathgit - This is working fine on my local. 
@Sanjeevani19 , @timur-isangulov  - Can you confirm if you are also getting this error or not ?

**Comment by Sanjeevani19** [comment]
> > With the latest code, I get this when I explore "Customer" entity
> > ```
> > react-dom.development.js:13413 Uncaught Error: Objects are not valid as a React child (found: object with keys {undefined}). If you meant to render a collection of children, use an array instead.
> >     in div (at results-tabular-view.tsx:140)
> >     in Trigger (created by Tooltip)
> >     in Tooltip (created by Context.Consumer)
> >     in Tooltip (created by MLTooltip)
> >     in MLTooltip (at results-tabular-view.tsx:138)
> >     in td (created by TableCell)
> >     in TableCell (created by TableRow)
> >     in tr (created by BodyRow)
> >     in BodyRow (created by TableRow)
> >     in TableRow (created by Connect(TableRow))
> >     in Connect(TableRow) (created by ExpandableRow)
> >     in ExpandableRow (created by Connect(ExpandableRow))
> >     in Connect(ExpandableRow) (created by BaseTable)
> >     in tbody (created by BaseTable)
> >     in table (created by BaseTable)
> >     in BaseTable (created by Connect(BaseTable))
> >     in Connect(BaseTable) (created by BodyTable)
> >     in div (created by BodyTable)
> >     in BodyTable (created by ExpandableTable)
> >     in div (created by ExpandableTable)
> >     in div (created by ExpandableTable)
> >     in ExpandableTable (created by Connect(ExpandableTable))
> >     in Connect(ExpandableTable) (created by Table)
> >     in Provider (created by Table)
> >     in Table (created by LocaleReceiver)
> >     in LocaleReceiver (created by Context.Consumer)
> >     in div (created by Context.Consumer)
> >     in div (created by Context.Consumer)
> >     in Spin (created by Context.Consumer)
> >     in div (created by Context.Consumer)
> >     in Table (created by withStore(Table))
> >     in withStore(Table) (created by MLTable)
> >     in MLTableContextProvider (created by MLTable)
> >     in MLTable (at results-tabular-view.tsx:327)
> >     in div (at results-tabular-view.tsx:326)
> >     in ResultsTabularView (at Browse.tsx:272)
> >     in div (at Browse.tsx:271)
> >     in div (at Browse.tsx:269)
> >     in main (created by Basic)
> >     in Basic (created by Context.Consumer)
> >     in Content (at Browse.tsx:231)
> >     in section (created by BasicLayout)
> >     in BasicLayout (created by Context.Consumer)
> >     in Layout (at Browse.tsx:221)
> >     in Browse (created by Context.Consumer)
> >     in withRouter(Browse) (at TilesView.tsx:34)
> >     in div (created by InternalMosaicWindow)
> >     in div (created by InternalMosaicWindow)
> >     in InternalMosaicWindow (created by Context.Consumer)
> >     in DragSource(InternalMosaicWindow) (created by Context.Consumer)
> >     in DropTarget(DragSource(InternalMosaicWindow)) (created by MosaicWindow)
> >     in MosaicWindow (at tiles.tsx:134)
> >     in div (created by MosaicRoot)
> >     in div (created by MosaicRoot)
> >     in MosaicRoot (created by MosaicWithoutDragDropContext)
> >     in div (created by MosaicWithoutDragDropContext)
> >     in MosaicWithoutDragDropContext (created by Mosaic)
> >     in Unknown (created by Mosaic)
> >     in Mosaic (at tiles.tsx:131)
> >     in Tiles (at TilesView.tsx:110)
> >     in div (at TilesView.tsx:108)
> >     in TilesView (at App.tsx:118)
> >     in Route (at App.tsx:34)
> >     in PrivateRoute (at App.tsx:117)
> >     in ModelingProvider (at App.tsx:98)
> >     in SearchProvider (at App.tsx:91)
> >     in Switch (at App.tsx:86)
> >     in div (at App.tsx:85)
> >     in main (at App.tsx:84)
> >     in div (at App.tsx:81)
> >     in App (created by Context.Consumer)
> >     in withRouter(App) (at src/index.tsx:12)
> >     in UserProvider (at src/index.tsx:11)
> >     in Router (created by BrowserRouter)
> >     in BrowserRouter (at src/index.tsx:10)
> >     at throwOnInvalidObjectType (react-dom.development.js:13413)
> >     at createChild (react-dom.development.js:13651)
> >     at reconcileChildrenArray (react-dom.development.js:13900)
> >     at reconcileChildFibers (react-dom.development.js:14305)
> >     at reconcileChildren (react-dom.development.js:16762)
> >     at updateHostComponent (react-dom.development.js:17302)
> >     at beginWork (react-dom.development.js:18627)
> >     at HTMLUnknownElement.callCallback (react-dom.development.js:188)
> >     at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
> >     at invokeGuardedCallback (react-dom.development.js:292)
> >     at beginWork$1 (react-dom.development.js:23203)
> >     at performUnitOfWork (react-dom.development.js:22157)
> >     at workLoopSync (react-dom.development.js:22130)
> >     at performSyncWorkOnRoot (react-dom.development.js:21756)
> >     at react-dom.development.js:11089
> >     at unstable_runWithPriority (scheduler.development.js:653)
> >     at runWithPriority$1 (react-dom.development.js:11039)
> >     at flushSyncCallbackQueueImpl (react-dom.development.js:11084)
> >     at flushSyncCallbackQueue (react-dom.development.js:11072)
> >     at discreteUpdates$1 (react-dom.development.js:21893)
> >     at discreteUpdates (react-dom.development.js:806)
> >     at dispatchDiscreteEvent (react-dom.development.js:4168)
> > ```
> 
> @srinathgit - This is working fine on my local.
> @Sanjeevani19 , @timur-isangulov - Can you confirm if you are also getting this error or not ?

@xnikhil08 this works fine in my local too.

**Comment by timur-isangulov** [comment]
> > > With the latest code, I get this when I explore "Customer" entity
> > > ```
> > > react-dom.development.js:13413 Uncaught Error: Objects are not valid as a React child (found: object with keys {undefined}). If you meant to render a collection of children, use an array instead.
> > >     in div (at results-tabular-view.tsx:140)
> > >     in Trigger (created by Tooltip)
> > >     in Tooltip (created by Context.Consumer)
> > >     in Tooltip (created by MLTooltip)
> > >     in MLTooltip (at results-tabular-view.tsx:138)
> > >     in td (created by TableCell)
> > >     in TableCell (created by TableRow)
> > >     in tr (created by BodyRow)
> > >     in BodyRow (created by TableRow)
> > >     in TableRow (created by Connect(TableRow))
> > >     in Connect(TableRow) (created by ExpandableRow)
> > >     in ExpandableRow (created by Connect(ExpandableRow))
> > >     in Connect(ExpandableRow) (created by BaseTable)
> > >     in tbody (created by BaseTable)
> > >     in table (created by BaseTable)
> > >     in BaseTable (created by Connect(BaseTable))
> > >     in Connect(BaseTable) (created by BodyTable)
> > >     in div (created by BodyTable)
> > >     in BodyTable (created by ExpandableTable)
> > >     in div (created by ExpandableTable)
> > >     in div (created by ExpandableTable)
> > >     in ExpandableTable (created by Connect(ExpandableTable))
> > >     in Connect(ExpandableTable) (created by Table)
> > >     in Provider (created by Table)
> > >     in Table (created by LocaleReceiver)
> > >     in LocaleReceiver (created by Context.Consumer)
> > >     in div (created by Context.Consumer)
> > >     in div (created by Context.Consumer)
> > >     in Spin (created by Context.Consumer)
> > >     in div (created by Context.Consumer)
> > >     in Table (created by withStore(Table))
> > >     in withStore(Table) (created by MLTable)
> > >     in MLTableContextProvider (created by MLTable)
> > >     in MLTable (at results-tabular-view.tsx:327)
> > >     in div (at results-tabular-view.tsx:326)
> > >     in ResultsTabularView (at Browse.tsx:272)
> > >     in div (at Browse.tsx:271)
> > >     in div (at Browse.tsx:269)
> > >     in main (created by Basic)
> > >     in Basic (created by Context.Consumer)
> > >     in Content (at Browse.tsx:231)
> > >     in section (created by BasicLayout)
> > >     in BasicLayout (created by Context.Consumer)
> > >     in Layout (at Browse.tsx:221)
> > >     in Browse (created by Context.Consumer)
> > >     in withRouter(Browse) (at TilesView.tsx:34)
> > >     in div (created by InternalMosaicWindow)
> > >     in div (created by InternalMosaicWindow)
> > >     in InternalMosaicWindow (created by Context.Consumer)
> > >     in DragSource(InternalMosaicWindow) (created by Context.Consumer)
> > >     in DropTarget(DragSource(InternalMosaicWindow)) (created by MosaicWindow)
> > >     in MosaicWindow (at tiles.tsx:134)
> > >     in div (created by MosaicRoot)
> > >     in div (created by MosaicRoot)
> > >     in MosaicRoot (created by MosaicWithoutDragDropContext)
> > >     in div (created by MosaicWithoutDragDropContext)
> > >     in MosaicWithoutDragDropContext (created by Mosaic)
> > >     in Unknown (created by Mosaic)
> > >     in Mosaic (at tiles.tsx:131)
> > >     in Tiles (at TilesView.tsx:110)
> > >     in div (at TilesView.tsx:108)
> > >     in TilesView (at App.tsx:118)
> > >     in Route (at App.tsx:34)
> > >     in PrivateRoute (at App.tsx:117)
> > >     in ModelingProvider (at App.tsx:98)
> > >     in SearchProvider (at App.tsx:91)
> > >     in Switch (at App.tsx:86)
> > >     in div (at App.tsx:85)
> > >     in main (at App.tsx:84)
> > >     in div (at App.tsx:81)
> > >     in App (created by Context.Consumer)
> > >     in withRouter(App) (at src/index.tsx:12)
> > >     in UserProvider (at src/index.tsx:11)
> > >     in Router (created by BrowserRouter)
> > >     in BrowserRouter (at src/index.tsx:10)
> > >     at throwOnInvalidObjectType (react-dom.development.js:13413)
> > >     at createChild (react-dom.development.js:13651)
> > >     at reconcileChildrenArray (react-dom.development.js:13900)
> > >     at reconcileChildFibers (react-dom.development.js:14305)
> > >     at reconcileChildren (react-dom.development.js:16762)
> > >     at updateHostComponent (react-dom.development.js:17302)
> > >     at beginWork (react-dom.development.js:18627)
> > >     at HTMLUnknownElement.callCallback (react-dom.development.js:188)
> > >     at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
> > >     at invokeGuardedCallback (react-dom.development.js:292)
> > >     at beginWork$1 (react-dom.development.js:23203)
> > >     at performUnitOfWork (react-dom.development.js:22157)
> > >     at workLoopSync (react-dom.development.js:22130)
> > >     at performSyncWorkOnRoot (react-dom.development.js:21756)
> > >     at react-dom.development.js:11089
> > >     at unstable_runWithPriority (scheduler.development.js:653)
> > >     at runWithPriority$1 (react-dom.development.js:11039)
> > >     at flushSyncCallbackQueueImpl (react-dom.development.js:11084)
> > >     at flushSyncCallbackQueue (react-dom.development.js:11072)
> > >     at discreteUpdates$1 (react-dom.development.js:21893)
> > >     at discreteUpdates (react-dom.development.js:806)
> > >     at dispatchDiscreteEvent (react-dom.development.js:4168)
> > > ```
> > 
> > 
> > @srinathgit - This is working fine on my local.
> > @Sanjeevani19 , @timur-isangulov - Can you confirm if you are also getting this error or not ?
> 
> @xnikhil08 this works fine in my local too.

Also works for me using reference-entity and hc-qa-project projects.

**Comment by xnikhil08** [comment]
@Sanjeevani19 , @timur-isangulov  , @srinathgit - Please verify if the file entity-search.lib is updated with the latest code as given in the PR. I just verified that if you don't comment out the maven-snapshot line from the build.gradle in project hc-qa-project or if your own project is using a maven snapshot then it probably still gets the latest snapshot code which might not include your local maven changes, which causes it to show the 500 error.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs [hunk 1]
```diff
@@ -365,6 +367,9 @@ function addGenericEntityProperties(result) {
     } else {
         addPrimaryKeyToResult(result, entityInstance, entityDef);
     }
+    } else {
+      console.log(`Unable to obtain a valid entity instance from document with URI '${result.uri}'; will not add entity properties to its search result`);
```
Comment (srinathgit) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs:371]: Please change this to xdmp.log()

Comment (srinathgit) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs:371]: ok, I didn't realize that

Comment (xnikhil08) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs:371]: @srinathgit - Its being used in few other places as well in the same file. If xdmp.log makes more sense and is more helpful then should we replace those as well ?

Comment (srinathgit) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs:371]: console.log should be ok , there doesn't seem to be a difference

**Pull request merged**